add `script` and `event` options

wkentdag opened the pull request:
This PR adds optional `--script` and `--event` flags, enabling users to specify the type of event to listen to, and to pass an optional script that gets triggered with the changed file `path` and `event`.

Ex: `"next-remote-watch ./content/**/ --script ./scripts/db/sync.js"`

Happy to build out tests for these flags as well!

hashicorp-cla commented:
[![CLA assistant check](https://cla.hashicorp.com/pull/badge/not_signed)](https://cla.hashicorp.com/hashicorp/next-remote-watch?pullRequest=1)


Thank you for your submission! We require that all contributors sign our Contributor License Agreement ("CLA") before we can accept the contribution. [Read and sign the agreement](https://cla.hashicorp.com/hashicorp/next-remote-watch?pullRequest=1)

[Learn more about why HashiCorp requires a CLA and what the CLA includes](https://www.hashicorp.com/cla)

<sub>Have you signed the CLA already but the status is still pending? [Recheck it](https://cla.hashicorp.com/check/hashicorp/next-remote-watch?pullRequest=1).</sub>

wkentdag commented:
@jescalan added some tests following the manual format you provided. I also renamed `options.script` to `options.command` and changed the interface so you can run any command, not just a node script.

jescalan commented on README.md:
```diff
@@ -7,10 +7,11 @@ This script replaces `next dev` with a version that adds a couple extra features
 "scripts": {
-  "start": "next-remote-watch" // this used to be "next dev"
+-  "start": "next dev"
++  "start": "next-remote-watch"
```
now this, i can get behind 🏆

kylemac commented on bin/next-remote-watch:
```diff
@@ -21,8 +27,25 @@ const handle = app.getRequestHandler()
+          console.error(e)
+          return e
+        }
+      }
```
i think we need to chain our `command` execution to the `reloadPage` right? bc otherwise if that hangs (say you forget a return in your `command` script) then you'd still get the `reloadPage` event. 

i'm not totally confident so check me on this.

jescalan replied:
It's inside a try/catch block so we're clear on this 😀

kylemac replied:
wouldn't that only protect us if it errors and not if it hangs (or just generally doesn't return something to `stderr`) ? and wouldn't it also not guarantee the next logs wouldn't be like 

```
[ wait ]  compiling ...
[ ready ] compiled successfully
Remote command failed
```

...which would be sort of confusing behavior. I would expect to see...

```
[ wait ]  compiling ...
Remote command failed
```

jescalan replied:
if it hangs i think we're in trouble regardless, but i see what you're saying -- you want the reload not to happen unless the command is successful? technically i don't see any reason to make the two dependent though -- we could have the site compile successfully and the remote command fail right?

kylemac replied:
yeah, i'm not 100% sure how i think it should be, tbh. i can go either way.

right now, i'm just thinking about learn authors who will be completely blind to the idea that there's a synchronization script happening and will be confused when the log says `compiled successfully` but the content isn't there. 

anyways, we'll use it as-is and see how it feels. i just wanted to raise it for our consideration.
